pkg/client/resourceaccessreview.go

```diff
@@ -19,42 +16,39 @@ type ResourceAccessReviewInterface interface {
 	Create(policy *authorizationapi.ResourceAccessReview) (*authorizationapi.ResourceAccessReviewResponse, error)
 }
 
-// resourceAccessReviews implements ResourceAccessReviewsNamespacer interface
+// resourceAccessReviews implements ResourceAccessReviews interface
 type resourceAccessReviews struct {
-	r  *Client
-	ns string
+	r *Client
 }
 
 // newResourceAccessReviews returns a resourceAccessReviews
-func newResourceAccessReviews(c *Client, namespace string) *resourceAccessReviews {
+func newResourceAccessReviews(c *Client) *resourceAccessReviews {
 	return &resourceAccessReviews{
-		r:  c,
-		ns: namespace,
+		r: c,
 	}
 }
 
-// Create creates new policy. Returns the server's representation of the policy and error if one occurs.
-func (c *resourceAccessReviews) Create(policy *authorizationapi.ResourceAccessReview) (result *authorizationapi.ResourceAccessReviewResponse, err error) {
+func (c *resourceAccessReviews) Create(rar *authorizationapi.ResourceAccessReview) (result *authorizationapi.ResourceAccessReviewResponse, err error) {
 	result = &authorizationapi.ResourceAccessReviewResponse{}
-	err = c.r.Post().Namespace(c.ns).Resource("resourceAccessReviews").Body(policy).Do().Into(result)
-	return
-}
 
-// clusterResourceAccessReviews implements ClusterResourceAccessReviews interface
-type clusterResourceAccessReviews struct {
-	r *Client
-}
+	// if this a cluster RAR, then no special handling
+	if len(rar.Action.Namespace) == 0 {
+		err = c.r.Post().Resource("resourceAccessReviews").Body(rar).Do().Into(result)
+		return
+	}
 
-// newClusterResourceAccessReviews returns a clusterResourceAccessReviews
-func newClusterResourceAccessReviews(c *Client) *clusterResourceAccessReviews {
-	return &clusterResourceAccessReviews{
-		r: c,
+	err = c.r.Post().Resource("resourceAccessReviews").Body(rar).Do().Into(result)
+
+	// if the namespace values don't match then we definitely hit an old server.  If we got a forbidden, then we might have hit an old server
+	// and shoudl try the old endpoint
+	if (rar.Action.Namespace != result.Namespace) || kapierrors.IsForbidden(err) {
+		deprecatedResponse := &authorizationapi.ResourceAccessReviewResponse{}
+		deprecatedAttemptErr := c.r.Post().Namespace(rar.Action.Namespace).Resource("resourceAccessReviews").Body(rar).Do().Into(deprecatedResponse)
+		if deprecatedAttemptErr == nil {
```

Comment:
if deprecatedAttemptErr is non-nil, we'll return the original result and err, which could be wrong in the `rar.Action.Namespace != result.Namespace` case...